pkg/pool-utils/contracts/ProtocolFeeCache.sol

```diff
@@ -126,6 +128,8 @@ abstract contract ProtocolFeeCache is RecoveryMode {
         _updateProtocolFeeCache(_protocolFeeProvider, ProtocolFeeType.AUM);
     }
 
+    function _beforeProtocolFeeCacheUpdate() internal virtual {}
```

Comment:
Lets add some natspec here explaining that we need to pay any protocol fees before updating the cache. This should be pretty similar to some of the comments you have elsewhere in the PR.